flink-libraries/flink-table/src/main/scala/org/apache/flink/table/descriptors/TableDescriptor.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.descriptors
+
+import org.apache.flink.table.descriptors.DescriptorProperties.toScala
+import org.apache.flink.table.descriptors.StatisticsValidator.{STATISTICS_COLUMNS, STATISTICS_ROW_COUNT, readColumnStats}
+import org.apache.flink.table.plan.stats.TableStats
+
+import scala.collection.JavaConverters._
+
+/**
+  * Common class for all descriptors describing table sources and sinks.
+  */
+abstract class TableDescriptor extends Descriptor {
+
+  protected var connectorDescriptor: Option[ConnectorDescriptor] = None
+  protected var formatDescriptor: Option[FormatDescriptor] = None
+  protected var schemaDescriptor: Option[Schema] = None
+  protected var statisticsDescriptor: Option[Statistics] = None
```

Comment:
I'm wondering if we really need these changes. A sink will never have a schema or statistics.

Reply:
I was wondering if a sink could have its own "schema configurations" for alignment with the output table schema? 
For example a CassandraTableSink / JDBCTableSink would definitely throw exceptions when trying to execute an insert with mismatched schemas.

Reply:
Thanks for pointing this out. You are right. Sinks can have a schema but no statistics. I was just wondering if we really need most of the refactorings in this PR. We need to rework the `TableSourceDescriptor` class in the near future because a Java user can access all `protected` field which is not very nice API design.

Reply:
Thanks a lot, Timo. We can consider refactoring the code later. How about simply moving statisticsDescriptor to SourceTableDescriptor for now?